packages/core/src/browser/connection-status-service.ts

```diff
@@ -62,92 +50,42 @@ export enum ConnectionState {
     ONLINE,
 
     /**
-     * The connection is lost between the client and the endpoint.
-     */
-    OFFLINE,
-
-    /**
-     * Initially we don't know whether we are online or offline.
+     * The connection is lost between frontend and backend.
      */
-    INITIAL
+    OFFLINE
 }
 
 @injectable()
 export class ConnectionStatusOptions {
 
     static DEFAULT: ConnectionStatusOptions = {
-        retry: 5,
-        retryInterval: 1000,
-        requestTimeout: 1000,
-        maxRetryInterval: 10000
+        offlineTimeout: 5000,
     };
 
     /**
-     * Number of accepted timeouts. Must be a positive integer.
-     */
-    readonly retry: number;
-
-    /**
-     * Retry interval in milliseconds. Must be a positive integer.
+     * Timeout in milliseconds before the application is considered offline. Must be a positive integer.
      */
-    readonly retryInterval: number;
+    readonly offlineTimeout: number;
 
-    /**
-     * The maximum retry interval in milliseconds. Should be a positive integer.
-     *
-     * If the request is timing out because of the slow Internet connection or the server is overloaded, we increase the `retryInterval` until it reaches this `maxRetryInterval`.
-     */
-    readonly maxRetryInterval: number;
-
-    /**
-     * Timeout for the HTTP GET request in milliseconds. Must be a positive integer.
-     */
-    readonly requestTimeout: number;
+}
 
+export const PingService = Symbol('PingService');
+export interface PingService {
+    ping(): Promise<void>;
 }
 
 @injectable()
-export class FrontendConnectionStatusService implements ConnectionStatusService, FrontendApplicationContribution {
+export abstract class AbstractConnectionStatusService implements ConnectionStatusService {
 
-    protected readonly statusChangeEmitter: Emitter<ConnectionStatus>;
-    protected readonly aliveUrl: string;
+    protected readonly statusChangeEmitter = new Emitter<ConnectionState>();
 
-    protected connectionState: ConnectionStatusImpl;
+    protected connectionState: ConnectionState = ConnectionState.ONLINE;
     protected timer: number | undefined;
-    protected retryInterval: number;
 
     constructor(
         @inject(ConnectionStatusOptions) @optional() protected readonly options: ConnectionStatusOptions = ConnectionStatusOptions.DEFAULT,
         @inject(ILogger) protected readonly logger: ILogger
     ) {
-        this.statusChangeEmitter = new Emitter<ConnectionStatus>();
```

Comment:
Can you please switch to property injection then?

Reply:
I want to use it without injection (see mock-connection-service.ts)

Reply:
I see, then it makes sense. 👍